examples/pytorch/rgcn/entity_classify_mp.py

```diff
@@ -466,34 +502,30 @@ def main(args, devices):
         else:
             val_idx = train_idx
 
-    # calculate norm for each edge type and store in edge
-    if args.global_norm is False:
-        for canonical_etype in hg.canonical_etypes:
-            u, v, eid = hg.all_edges(form='all', etype=canonical_etype)
-            _, inverse_index, count = th.unique(v, return_inverse=True, return_counts=True)
-            degrees = count[inverse_index]
-            norm = th.ones(eid.shape[0]) / degrees
-            norm = norm.unsqueeze(1)
-            hg.edges[canonical_etype].data['norm'] = norm
+    node_feats = []
+    for ntype in hg.ntypes:
+        if len(hg.nodes[ntype].data) == 0 or args.node_feats is False:
+            node_feats.append(hg.number_of_nodes(ntype))
+        else:
+            assert len(hg.nodes[ntype].data) == 1
+            feat = hg.nodes[ntype].data.pop('feat')
+            node_feats.append(feat.share_memory_())
 
     # get target category id
     category_id = len(hg.ntypes)
     for i, ntype in enumerate(hg.ntypes):
         if ntype == category:
             category_id = i
 
-    g = dgl.to_homogeneous(hg, edata=['norm'])
-    if args.global_norm:
-        u, v, eid = g.all_edges(form='all')
-        _, inverse_index, count = th.unique(v, return_inverse=True, return_counts=True)
-        degrees = count[inverse_index]
-        norm = th.ones(eid.shape[0]) / degrees
-        norm = norm.unsqueeze(1)
-        g.edata['norm'] = norm
-
-    g.ndata[dgl.NTYPE].share_memory_()
-    g.edata[dgl.ETYPE].share_memory_()
+    g = dgl.to_homogeneous(hg)
+    gen_norm(g)
     g.edata['norm'].share_memory_()
+    g.ndata['ntype'] = g.ndata[dgl.NTYPE]
+    g.ndata['ntype'].share_memory_()
+    g.edata['etype'] = g.edata[dgl.ETYPE]
+    g.edata['etype'].share_memory_()
+    g.ndata['type_id'] = g.ndata[dgl.NID]
+    g.ndata['type_id'].share_memory_()
```

Comment:
why do you want to rename them?

Reply:
I think it is not recommended to use buildin names for features.